ably.d.ts

```diff
@@ -122,7 +122,7 @@ declare namespace Types {
 		 * Can be used to explicitly recover a connection.
 		 * See https://www.ably.com/documentation/realtime/connection#connection-state-recovery
 		 */
-		recover?: string | ((lastConnectionDetails: {
+		recover?: boolean | string | ((lastConnectionDetails: {
```

Comment:
Setting `options.recover` as a boolean is deprecated [see here](https://github.com/ably/ably-js/blob/main/common/lib/util/defaults.js#L144). I'm not sure why this change was needed here but you should be able to make a similar change to `DeprecatedClientOptions` and get the same effect.

Reply:
Changing the signature of recover on `DeprecatedClientOptions` causes `ClientOptions` to no longer be compatible with it, meaning functions that take `ClientOptions` instead of `DeprecatedClientOptions` throw an error instead: 

```
Argument of type 'DeprecatedClientOptions' is not assignable to parameter of type 'ClientOptions'.   
Types of property 'recover' are incompatible.
```

Reply:
Ah that's annoying. Unfortunately it's not really an option to make this change to the public API since setting `options.recover` to a boolean is deprecated. Could you look in to alternatives? I'm happy for you to use a workaround like `any` just to get this over the line and then we can deal with it later, just need to make sure we're not changing the public type interface.

Reply:
Got it, there's only one place where this is checked, so [I've modified that](https://github.com/ably/ably-js/pull/890/files#diff-915197b352a9482f9518c50eb5909c5820d931511a1aeb8a21836e32c04b1d40R178) to convert to `any` first